### ISSUE TITLE
Question about Docker

### ISSUE THREAD

**Issue by dvdavide** [deiger/AirCon#64]
First off, I **love** Docker. Makes upgrading so much easier.
As I was saying in the last issue I opened, I have AEH-W4B1 and AEH-W4E1 modules, which need separate apps to control them.

Without having to rewrite the docker package for people in my situation, I was wondering if it was possible to put the config files (obtained the non-docker way) in the config directory and then run the docker package. Can I try or you're already sure it won't work? Or do I need to set up two separate Docker instances? Thanks!

**Comment by deiger** [comment]
At the moment the setup is limited to a single app, though naturally you can edit the config file manually.
That said, assuming you're OK with having the same credentials for both apps, it should be easy to change the Dockerfile to accommodate a list of apps. @rotem925 added the docker support.

**Comment by rotem925** [comment]
In this case I would actually run the container twice for each app. The app and the credentials only needed for the discovery.
So just run it twice for each app. This will generate config files.
If you already have the config files, just copy them over to the config directory/volume.
@deiger can you confirm this please?

**Comment by deiger** [comment]
I think it would be cleaner if we just run the discovery twice, once for each app, and collect all the config files. The server supports ACs from multiple apps, and it will be difficult to have multiple containers when running inside HA.

**Comment by deiger** [comment]
If we can assume the credentials for the apps are the same, it should be easy to support. Alternatively we can also make username and password repeated.

**Comment by rotem925** [comment]
> I think it would be cleaner if we just run the discovery twice, once for each app, and collect all the config files. The server supports ACs from multiple apps, and it will be difficult to have multiple containers when running inside HA.

That exactly what I meant. 👍 
After the discovery, the server can run with all the config files as a single running server

**Comment by rotem925** [comment]
> If we can assume the credentials for the apps are the same, it should be easy to support. Alternatively we can also make username and password repeated.

I wouldn't assume the username/passwords are the same...
I think running the discovery twice will be the best and cleaner.

**Comment by rotem925** [comment]
Another option would be to run with environment variable
for example, RUN_MODE=server/discovery
So then, we can only generate the config files, regardless of the folder content, and of course not deleting them..

**Comment by dvdavide** [comment]
So I tried setting up the container and copied over the config files onto the /config designated folder **before** (not sure if relevant, but probably worth mentioning) starting it, and everything seems to be working fine.

Had to edit the last line of the docker-compose like this, putting a slash in front of config:
`- /config:/home/<user>/hisense`

I'll leave the issue open in case anyone has suggestions on how to implement it in a setup from scratch, but @deiger feel free to close it as you see fit.

Thanks again everyone.

**Comment by deiger** [comment]
I wouldn't make a separate run, as it wouldn't play well with [this](https://github.com/deiger/AirCon/blob/master/hassio/config.json).
The config for HA supports nested options, not sure how they manifest in ENV.

**Comment by dvdavide** [comment]
> So I tried setting up the container and copied over the config files onto the /config designated folder **before** (not sure if relevant, but probably worth mentioning) starting it, and everything seems to be working fine.

Actually I was over-enthusiastic, I'm afraid. The container only works for the machines that respond to the app specified in the config. When I saw the response from the curl request and the log file scrolling along nicely I thought everything was working, but I was wrong.
Recreated the container specifying the other app, and then the other AC showed up but the others had disappeared.

**Comment by dvdavide** [comment]
So I tried making separate "hisense_W4B1" and "hisense_W4E1" containers, with different credentials and different port, and they seem to live side by side fine in the few minutes I've had them up. I hope I'm not wrong again...

**Comment by dvdavide** [comment]
Last update: corrected the docker-compose to:
`- /home/<user>/hisense:/config `
as per issue #65 
(from within Portainer it was all much clearer)
and now I see all the machines from one instance.

**Comment by deiger** [comment]
I've changed the docker logic to get a config file with the definitions for multiple apps. Please try it out.
Also, there is now support for doing the setup from within HA supervisor, with the same config.

**Comment by dvdavide** [comment]
I tried it, and the copy_config didn't work for me. I kept getting this error:

```
Building copy_config
ERROR: Cannot locate specified Dockerfile: Dockerfile
```

However, I just manually copied over the options.json file to the config folder, removed lines 3-9 and 11-12 from the dockerfile,  tried again, and it worked.

In the options.json I assumed I only had to repeat lines 3-7 for each different app, and separate the { } sections with a comma. Seems to be working fine.

First attempt was pointing the container to the config dir which already had the config files from the previous installation and it worked. 
Second attempt was with config dir pointing at a new dir (empty except for options.json) and it also worked, created config files no problem.
